src/main/java/ar/com/synergian/wagongit/Utils.java

```diff
@@ -11,15 +11,14 @@
 	private Utils() {
 	}
 
-	public static File createCheckoutDirectory(String path, boolean isPermanent) throws GitException {
+	public static File createCheckoutDirectory(String path) throws GitException {
 		File dir;
-		if (isPermanent) {
-			dir = new File(path);
-		} else {
+		if (path.endsWith(".git")) {
 			dir = new File(System.getProperty("java.io.tmpdir"), "wagon-git-" + hashPath(path));
+		} else {
+			dir = new File(path);
```

Comment:
I'm sorry. I'm messing things up. I wrote that part of the code like 4 or 5 years ago. Forget that. Although, the name of the property should be `wagon.git.cleanup.directory` instead of `wagon.git.safe.checkout`. I probably changed that at some point. I honestly don't remember.

Change that if you want. I'm okay if you don't.

I'll just await for the documentation of the two new flags.